uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/calendar-ctrl.js

```diff
@@ -326,6 +328,15 @@ define([
     };
 
     /**
+     * Reset the months data for before refresh
+     */
+    this._resetMonths = function () {
```

Comment:
1. We have a structure like this
```
{
  month: startDate.month(),
  year: startDate.year(),
  data: []
}
```

2. It is necessary because when the calendar reloads because of a leave request is created/edited the `data` should clear before reassigning values, otherwise duplicate values are shown in the UI.